### ISSUE TITLE
Transitive groups as lazy artifact? 

### ISSUE THREAD

**Issue by lgoettgens** [oscar-system/Oscar.jl#2850]
Similar to the perfect groups library from GAP, the `transgrp` package doesn't ship all its files but instead requires the user to download them separately and put the into a specific folder in the gap installation.
For the perfect groups library this was solved for oscar by using lazy artifacts that downloaded the required files on first use. Do we want something similar for `transgrp`?
If yes, I can look into adding this. 

One important thing to consider: transgrp has much larger files, one of 300MB and one of 30GB.

Cc @fingolfin @ThomasBreuer

**Comment by micjoswig** [comment]
This should be discussed within the upcoming B1 proposal.

**Comment by fingolfin** [comment]
I don't think that the huge transgrp files are a good fit for *automatic* downloading, at least not without giving users the chance to accept/reject this. For this it'd be much better if only those parts of this "database" that are actually needed could be loaded... So, if it indeed was more like a database, and not just a blob of data... So yeah, possibly something for the next B1 and/or MaRDI

**Comment by fingolfin** [comment]
The same applies to a recent extension of the database of primitive groups: it was extended from degree <= 4095 to degree <= 8191. The extension can be downloaded as a single tarball from Zenodo, but it actually consists of many small files. We'd want access  to this in a more granular fashion, I'd say.
